core/src/main/kotlin/fr/sncf/osrd/stdcm/infra_exploration/InfraExplorerWithEnvelopeImpl.kt

```diff
@@ -87,22 +86,20 @@ data class InfraExplorerWithEnvelopeImpl(
     }
 
     override fun withReplacedEnvelope(envelope: Envelope): InfraExplorerWithEnvelope {
-        return copy(
-            envelopes = appendOnlyLinkedListOf(LocatedEnvelopeInterpolate(envelope, 0.0, 0.0)),
-            spacingRequirementAutomaton =
-                SpacingRequirementAutomaton(
-                    spacingRequirementAutomaton.rawInfra,
-                    spacingRequirementAutomaton.loadedSignalInfra,
-                    spacingRequirementAutomaton.blockInfra,
-                    spacingRequirementAutomaton.simulator,
+        val spacingResourceGenerator =
```

Comment:
nit: for consistency
```suggestion
        val spacingRequirementAutomaton =
```